src/lib/formatter/SarifFormatter.ts

```diff
@@ -81,9 +84,10 @@ abstract class SarifFormatter {
 							text: description.trim().replace(/^\n/, '').replace(/\n$/, '')
 						},
 						properties: {
-							category: v.category,
-							severity: v.severity
-						}
+						category: v.category,
+						severity: v.severity,
+						normalizedSeverity: (this.normalizeSeverity ? v.normalizedSeverity : undefined)
```

Comment:
We should be checking before the `normalizedSeverity` field is added to the structure so that existing users don't see a difference. I suppose we'll need to build this json dynamically. Have a look at this: https://stackoverflow.com/questions/15009448/creating-a-json-dynamically-with-each-input-value-using-jquery

Also, we might have some internal code examples. Will update here if I can find it.

Reply:
Here's one from our code: https://github.com/gjung-sfdc/sfdx-scanner/blob/d4571b367f54c3624fdf5d3c9ff0fc0cea5b346a/src/lib/eslint/BaseEslintEngine.ts#L262

Instead of a foreach, you can set individual keys/values. And based on `normalizedSeverity` check, you can push the key/value for line 89.

Reply:
The normalizedSeverity field doesn't appear when it's set to undefined, only when it's set to a value.